test/units/modules/network/nxos/test_nxos_vpc.py

```diff
@@ -64,3 +80,82 @@ def test_nxos_vpc_present(self):
             'peer-keepalive destination 192.168.100.4 source 10.1.100.20',
             'peer-gateway', 'auto-recovery',
         ])
+
+    def test_nxos_vpc_vrf_1(self):
+        # No vrf -> vrf 'default'
+        set_module_args(dict(
+            domain=100,
+            pkl_dest='192.168.1.1',
+            pkl_src='10.1.1.1',
+            pkl_vrf='default',
+        ))
+        self.execute_module(changed=True, commands=[
+            'vpc domain 100',
+            'peer-keepalive destination 192.168.1.1 source 10.1.1.1 vrf default'
+        ])
+
+    def test_nxos_vpc_vrf_2(self):
+        # vrf 'my_vrf'-> vrf 'test-vrf'
+        # All pkl commands should be present
+        self.get_config.return_value = load_fixture('nxos_vpc', 'vrf_test_vpc_config')
+        set_module_args(dict(
+            domain=100,
+            pkl_dest='192.168.1.1',
+            pkl_src='10.1.1.1',
+            pkl_vrf='test-vrf',
+        ))
+        self.execute_module(changed=True, device='_vrf_test', commands=[
+            'vpc domain 100',
+            'peer-keepalive destination 192.168.1.1 source 10.1.1.1 vrf test-vrf'
+        ])
+
+    def test_nxos_vpc_vrf_3(self):
+        # vrf 'my_vrf'-> vrf 'test-vrf'
+        # Existing pkl_src should be retained even though playbook does not specify it
+        self.get_config.return_value = load_fixture('nxos_vpc', 'vrf_test_vpc_config')
+        set_module_args(dict(
+            domain=100,
+            pkl_dest='192.168.1.1',
+            pkl_vrf='test-vrf',
+        ))
+        self.execute_module(changed=True, device='_vrf_test', commands=[
+            'vpc domain 100',
+            'peer-keepalive destination 192.168.1.1 source 10.1.1.1 vrf test-vrf'
+        ])
+
+    def test_nxos_vpc_vrf_4(self):
+        # vrf 'my_vrf'-> vrf 'management'
+        # 'management' is the default value for vrf, it will not nvgen
+        self.get_config.return_value = load_fixture('nxos_vpc', 'vrf_test_vpc_config')
+        set_module_args(dict(
+            domain=100,
+            pkl_dest='192.168.1.1',
+            pkl_vrf='management',
+        ))
+        self.execute_module(changed=True, device='_vrf_test', commands=[
+            'vpc domain 100',
+            'peer-keepalive destination 192.168.1.1 source 10.1.1.1 vrf management'
+        ])
+
+    def test_nxos_vpc_vrf_5(self):
+        # vrf 'my_vrf' -> vrf 'my_vrf' (idempotence)
+        self.get_config.return_value = load_fixture('nxos_vpc', 'vrf_test_vpc_config')
+        set_module_args(dict(
+            domain=100,
+            pkl_dest='192.168.1.1',
+            pkl_src='10.1.1.1',
+            pkl_vrf='my_vrf',
+        ))
+        self.execute_module(changed=False, device='_vrf_test')
+
```

Comment:
I think you should also add the following test

```python
    def test_nxos_vpc_vrf_7(self):
        # vrf 'my_vrf' -> vrf 'my_vrf' (idempotence)
        self.get_config.return_value = load_fixture('nxos_vpc', 'vrf_test_vpc_config')
        set_module_args(dict(
            domain=100,
            pkl_dest='192.168.1.1',
            pkl_vrf='my_vrf',
        ))
        self.execute_module(changed=False, device='_vrf_test')
```

Reply:
done